pkg/volumes/openstack/volumes.go

```diff
@@ -352,13 +354,58 @@ func GetDevicePath(volumeID string) (string, error) {
 	return "", fmt.Errorf("Failed to find device for the volumeID: %q", volumeID)
 }
 
+// probeVolume probes volume in compute
+// see issue https://github.com/kubernetes/cloud-provider-openstack/issues/705
+func probeVolume() error {
+	// rescan scsi bus
+	scsiPath := "/sys/class/scsi_host/"
+	if dirs, err := ioutil.ReadDir(scsiPath); err == nil {
+		for _, f := range dirs {
+			name := scsiPath + f.Name() + "/scan"
+			data := []byte("- - -")
+			ioutil.WriteFile(name, data, 0666)
+		}
+	}
+
+	executor := utilexec.New()
+	args := []string{"trigger"}
+	cmd := executor.Command("udevadm", args...)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // FindMountedVolume implements Volumes::FindMountedVolume
 func (_ *OpenstackVolumes) FindMountedVolume(volume *volumes.Volume) (string, error) {
-	device, err := GetDevicePath(volume.ProviderID)
-	if err != nil {
-		return "", err
+	// wait for 2.5min is the volume attached and path found
```

Comment:
It does look like FindMountedVolume is itself called from a fairly tight loop in safeFormatAndMount.  I think the actual delta here is adding the scsi path probing and udevadm call.  I do wonder if other platforms would also benefit from similar code as well (and the idea of the smarter backoff is a good one, because safeFormatAndMount just waits a second in between polls...)

Oh... which makes me wonder if the actual problem here is that FindMountedVolume should be returning `"", nil` when it doesn't find the volume, it looks like it was returning `"", fmt.Errorf(..)` ?

Anyway, I'm in favor of getting this code in and seeing if we can maybe promote some of this into FindMountedVolume / other clouds.